ethers-core/src/utils/mod.rs

```diff
@@ -421,6 +421,27 @@ pub fn to_checksum(addr: &Address, chain_id: Option<u8>) -> String {
     })
 }
 
+/// checks if the given address is a valid checksum address
+pub fn verify_checksum(addr: &str, chain_id: Option<u8>) -> bool {
```

Comment:
don't love the name,
also not a fan of the `to_checksum` function, should have been named `to_checksum_address`
but, to keep it consistent, this is fine I guess.

Reply:
maybe `check_address_checksum`?
ref: https://github.com/web3/web3.js/blob/aaf26c8806bc9fb60cf6dcb6658104963c6c7fc7/packages/web3-utils/src/Utils.js#L140